task/mongo-optimization-task.js

```diff
@@ -243,22 +270,29 @@ async function task_3_1(db) {
         {
             "$lookup" : {
                 "from" : "clientCriteria",
-                "localField" : "criteria_value",
-                "foreignField" : "value",
+                let: { "crit_value": "$criteria_value"},
+                pipeline: [
+                {
+                    $match : 
+                    {
+                        "versions.initiativeId" : ObjectId("58af4da0b310d92314627290")
+                    }
+                },
+                {
+                    $project: {
+                        value: 1,
+                        label: 1,
+                        definition: 1,
+                        'versions.definition': 1,
+                    }
+                },                   
+                { 
+                    $match: { $expr: { $eq: [ "$value",  "$$crit_value" ] } }
```

Comment:
- you can make the match before projection above.
- it's possible to use limit: 1 here